Cargo.toml

```diff
@@ -21,10 +21,10 @@ stream = []
 log = "0.4"
 futures-util = { version = "0.3", default-features = false, features = ["async-await", "sink", "std"] }
 pin-project = "1.0"
-tokio = { version = "0.3", default-features = false, features = ["io-util"] }
+tokio = { version = "1.0.0", default-features = false, features = ["io-util"] }
```

Comment:
If semantic versioning is used properly for this, which it should be, couldn't this even be "1" as tokio is stable on 1.x.x?

Reply:
That's a valid point. I mostly also specify the minor version in my apps so i didn't think of that. So, the tokio version here should just be `1`

Reply:
Right, per https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements `1`, `1.0` and `1.0.0` are all equivalent.

Then it is a question of style/consistency. `1.0` would be most consistent locally (see line above). Is anybody aware of an ecosystem-wide style recommendation for equivalent version declarations in Cargo.toml?

Reply:
All I remember is Cargo endorses semantic versioning and the major version thats `1+.x.x` won't have any breaking changes. It's fine to use `1.0` as well for consistency and in case of local bugfixes that get added/removed due to tokio in this repo as tokio is developed/patched

Reply:
As @strohel mentioned 1.0.0 is equivalent to 1.x.x. the only exception to this rule are 0.x.x releases. For example 0.1.0 would be equivalent to 0.1.x.

I personally prefer the full version as it kind of is a statement to what version the project was tested with by the authors and considered to be working. So that later when an error happens only on the user side and the user states that he is using for example the auto upgraded 1.5.3 instead of 1.0.0, then it's easy to derive that the error lies in some changes of the said dependency. Or at least would be likely and a good thing to look at first.

In the end it's up to you folks from snapview. I just wanted to explain my reasoning behind putting the full version in the Cargo.toml.

Reply:
It's not pinned, this is equivalent to `^1.0.0` ([docs](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio)) and will allow updates to any `1.x.x`.

Reply:
> It's not pinned, this is equivalent to `^1.0.0` ([docs](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio)) and will allow updates to any `1.x.x`.

TIL. Thanks for informing me

Reply:
Ahh same, didn't know that was how it worked; I assumed by ommision of one of the versions is allowed it to be an `x`, so `1.0` would be `1.0.x`, guess TIL